### ISSUE TITLE
Research: select Laya or Kev-0.8B for the local semantic-decision lane

### ISSUE THREAD

**Issue by Unjuno** [Unjuno/agent-interface#4203]
## Motivation

Guidance: #4158. Parent direction: #9. Related typed-decision work: #853, #1014, #1024, #4153.

Agent Interface now has two credible local-only candidates for the same semantic-decision role:

- **Laya** — small non-autoregressive typed decision model, optimized for one-pass `choice / score / noul` decisions.
- **Kev-0.8B** — Qwen3.5-0.8B + LoRA + pointer head, also exposing `choice / score / noul`, with stronger published transfer/fine-tune machinery but more compute.

These are not intended to be placed serially in every control path. They compete for the same slot:

```text
current state + Astra intent + bounded candidates
    -> semantic local decision
    -> typed proposal / YIELD
```

The goal is to determine whether one should be the default local semantic-decision backend, or whether they justify distinct latency/capability envelopes.

Current-main intake for this Issue: `5542d80583616b204531e1eb590125a41704929c` (2026-09-23). Exact model/code revisions used by the experiment must be frozen separately before allocation.

Targeted duplicate search found no existing Laya-vs-Kev Issue. #853 owns pretrained tabular ICL, #1014 owns training-free shared-KV/readout mechanics, and #1024 owns the general planner-gap backend ladder. This Issue owns only the **matched Laya-vs-Kev-0.8B discriminator**.

## H — hypothesis

Under one frozen Agent Interface semantic-decision contract, either:

1. one of Laya or Kev-0.8B satisfies the required intent fidelity, YIELD behavior, and local latency/resource envelope strongly enough that keeping both as default backends adds no demonstrated value; or
2. the two occupy measurably distinct envelopes, e.g. Laya is materially faster on narrow/stable decisions while Kev-0.8B preserves materially more semantic/transfer coverage.

A difference in public benchmark score alone is not sufficient. The result must come from the same Agent Interface rows, hardware class, input contract, and scorer.

## T — smallest meaningful matched comparison

Authority-neutral shadow mode. No OS task input.

Freeze one typed decision vocabulary such as:

```text
CONTINUE
WATCH
REPAIR
YIELD
```

Freeze a common input schema containing only caller-visible information needed by both models:

```text
intent
constraints
current typed state
last verified effect where applicable
candidate descriptions
scope/version metadata
```

Use the same rows for both candidates, including:

1. **SAME_STATE_DIFFERENT_INTENT** — identical state, changed intent/constraints.
2. **SAME_INTENT_STATE_VARIATION** — held-out in-scope state changes.
3. **AMBIGUOUS_OR_MISSING_EVIDENCE** — expected YIELD.
4. **OUT_OF_SCOPE** — expected YIELD.
5. **OPTION_ORDER_PERTURBATION** — same semantics, permuted candidate order.
6. **SHORT_STATE / LONG_STATE** — frozen context-length buckets.
7. **REPEATED_STATE** — measure valid cache/reuse benefit separately from fresh-state performance.

Pin exact Laya checkpoint/revision and exact Kev-0.8B checkpoint/revision. Local inference only; no remote fallback. Any model-specific prompt/serialization adapter must be frozen before the formal allocation and may not add semantic information unavailable to the other arm.

Do not fine-tune either model in the first discriminator. Fine-tuned comparison is a separate successor because it changes a second factor.

## Primary measurements

Report separately, with no synthetic aggregate score:

- independently acceptable disposition/effect rate;
- same-state/different-intent correctness;
- held-out state-variation correctness;
- required-YIELD recall;
- unnecessary-YIELD rate;
- forbidden/disallowed proposal rate;
- option-order sensitivity;
- fresh-state p50/p95/p99/max;
- repeated-state p50/p95/p99/max;
- cold load / warm startup;
- peak RSS / VRAM;
- model/artifact bytes;
- preprocessing/tokenization time separately from model compute.

Confidence/calibration is diagnostic only. Confidence never becomes authority.

## D — decisions

Retain one of:

- `SELECT_LAYA_SEMANTIC_LANE_SCOPED` — Laya meets all frozen semantic/YIELD gates and Kev-0.8B provides no preregistered capability gain sufficient to justify its extra resource/latency cost.
- `SELECT_KEV_08B_SEMANTIC_LANE_SCOPED` — Kev-0.8B meets all gates and provides a preregistered semantic/transfer advantage while staying inside the declared local latency/resource envelope.
- `RETAIN_BOTH_DISTINCT_ENVELOPES_SCOPED` — both pass but occupy different frozen envelopes that are useful enough to justify routing.
- `FAIL_LAYA_SEMANTIC_OR_YIELD` / `FAIL_KEV_SEMANTIC_OR_YIELD` — candidate misses frozen semantic or abstention gates.
- `HOLD_NO_DISCRIMINATING_TASK` — the fixture does not separate the candidates.
- `STOP_MODEL_OR_PROVENANCE_UNAVAILABLE` — exact local weights/runtime/source cannot be frozen.

Selection is scoped to the tested lane and hardware. It is not a general model ranking.

## C — competing explanations

- Kev's larger base may only help because the fixture rewards broad language knowledge rather than the intended bounded decision contract.
- Laya may look better only because the task is narrow and close to its training/fine-tuning distribution.
- Repeated-state caching can dominate latency and may not transfer to rapidly changing GUI state.
- Serialization quality may explain apparent semantic differences.
- A simpler deterministic rule/tree may already solve the same contract.

## U — uncertainty / limits

- No live GUI authority or task-effect claim.
- No task-local fine-tuning claim in this Issue.
- Hardware-specific latency cannot be generalized across CPU/GPU/Apple Silicon.
- Public author benchmarks remain background only; this Issue requires one matched local comparison.
- If both fail, the next step is not automatically a larger model; representation sufficiency must be checked first.

## Promotion rule

Only the selected/scoped backend may become the default **candidate** for the semantic lane. Runtime promotion still requires a separate integrated action/effect experiment under ordinary deterministic authority.

**Comment by Unjuno** [comment]
Local execution-availability check (2026-09-23): before any #4203 allocation, the supplied Linux execution environment reports no `docker` or `podman` executable and a bounded read-only search of /root/.cache, /home/oai/.cache and /mnt/data found no Laya, Kev, or Qwen3.5-0.8B weight/cache path. No model call, download, branch, source freeze, or formal row was created.

This is a **local environment/provenance availability record**, not `STOP_MODEL_OR_PROVENANCE_UNAVAILABLE` for the repository as a whole and not evidence that the models are unavailable elsewhere. Per failure routing, do not create a wrapper-only successor. A real allocation should begin only in an authorized environment where exact candidate weights/runtime revisions can be frozen.
